### ISSUE TITLE
Coding philosophy (no use of document or window)

### ISSUE THREAD

**Issue by Sleashe** [vuejs/vue-hackernews-2.0#156]
Hi guys,

We're basing our new project on top of **this excellent hackernews biolerplate**. We're discovering how webpack and the server side rendering part are working

If I'm right, window or document object are not accessible inside of .vue files and I got a few questions about that : 

1. I've read a lot of things about server side rendering using Vue.js but I don't understand why the window object is not available after the rendering process ?

2. We would like to adopt THE _good_ coding philosophy (and avoid the use of jsdom hack), and it's not a problem to **not** use window or document object, but what's the **clean** workaround to do that ? 
For instance, if I would like to access window.document.cookies to manage my authentication token inside a **separate** api.js file (which will be imported into a .Vue file for instance), how should I do ?

3. I noticed that the window object is fully available from the _client-entry.js_ file ? I thought I could send a few needed _window_ values into the store using this file, again is this clean ? Or totally ugly ?

Thanks a lot in advance for your answers,

btw, VueJs, and the ecosystem are just awesome !

**Comment by yyx990803** [comment]
Wether you have access to `window` in a component depends on which **lifecycle hooks** your code is in.

Essentially, the only lifecycle hooks that are called on the server-side are `beforeCreate` and `created`. That means inside hooks like `mounted` you do have access to `window`.

**Comment by Sleashe** [comment]
Thanks a lot Evan for your ultra fast answer. Following your response, I tried, and I can access the window object inside `mounted` hook of a component. 👍 

I got one last question, if it doesn't disturb you:

I have a separate api.js file where I use axios to make requests. I would like to register an interceptor, to inject specific headers with a value found inside a cookie. I'm completly stuck right now, because I do not have access to the window object into this separate api.js file.

I found this post which could be helpful ([https://forum.vuejs.org/t/change-initial-state-for-ssr-client-hydration/3924/2](https://forum.vuejs.org/t/change-initial-state-for-ssr-client-hydration/3924/2)), but I would like to know your point of view, and your recommandations to implement this the right way.
